#163424172 create office

haywhyze opened the pull request:
#### What does this PR do?
Implement create a political office 
#### Description of Task to be completed?
- write tests for the functionalities of the create office endpoint
- add a controller class to handle `/offices` endpoint
- modify isEmpty middleware to accomodate `/offices`
- add new middlewares to handle user input.
- create a data model for the office entity.
- add the new controller and neccessary middlewares to the router
#### How should this be manually tested?
- clone this repository by executing `git clone https://github.com/haywhyze/politico.git`
- cd to the cloned local repository `cd politico`
- run `ft-create-office-163424172`
- install all dependencies by running `npm install`
- start the server with `npm start`
- when the server is up, you can test using Postman or any other related service.

#### What are the relevant pivotal tracker stories?
[#163424172](https://www.pivotaltracker.com/story/show/163424172)

haywhyze committed:
feat(server): implement get one party

- write test scripts to test the getOne party functionalilty
- create a middleware to validate party ID
- create getOne method to handle get request to `/parties/<party-id>`
- add the getOne method and the validateID middleware to the router

[delivers #163423238]

haywhyze committed:
feat(server): implement create office endpoint

- write tests for the functionalities of the create office endpoint
- add a controller class to handle `/offices` endpoint
- modify isEmpty middleware to accomodate `/offices`
- add new middlewares to handle user input.
- create a data model  for the office entity.
- add the new controller and neccessary middlewares to the router

[delivers #163424172]

haywhyze committed:
Update office.test.js

- update tests scripts to test for presence of error message when applicable.

[finishes #163424172]

akinwale-habibullah commented on server/middlewares/isEmpty.js:
```diff
@@ -1,13 +1,11 @@
-  if (!req.body.hqAddress) {
-    error.push('hqAddress');
-  }
+  // eslint-disable-next-line array-callback-return
```
Please use rules in .eslintrc file instead of specifying your rules in individual files.

haywhyze replied:
done!

haywhyze committed:
update eslintrc

- remove inline rules for global rules
[finishes #163424172]

akinwale-habibullah commented on server/middlewares/validateType.js:
```diff
@@ -0,0 +1,13 @@
+  if (!acceptedType.includes(req.body.type)) {
+    return res.status(400).send({
+      status: 400,
+      error: 'Type not of accepted value',
```
Please specify the allowed types, so you don't leave the user confused.

haywhyze replied:
okay

akinwale-habibullah commented:
Please implement my feedback, commit and then go ahead to merge.

haywhyze committed:
Update validateType.js

- update error message to indicate accepted types in order not to confuse the user

[delivers #163424172]
